File: src/main/java/mediathek/gui/dialogEinstellungen/allgemein/TimedDocumentListener.java
package mediathek.gui.dialogEinstellungen.allgemein;

import javax.swing.*;
import javax.swing.event.DocumentEvent;
import javax.swing.event.DocumentListener;
import java.awt.event.ActionListener;

/**
 * Listens to changes in a Document and finally fires the assigned Action.
 * All operations will be performed on Swing EDT.
 */
public final class TimedDocumentListener implements DocumentListener {
    private final Timer timer;

    public TimedDocumentListener(ActionListener taskPerformer) {
        timer = new Timer(1_000, taskPerformer);
        timer.setRepeats(false);
    }

    private void fireTimer() {
        if (timer.isRunning())
            timer.restart();
        else
            timer.start();
    }

    @Override
    public void insertUpdate(DocumentEvent e) {
        fireTimer();
    }

    @Override
    public void removeUpdate(DocumentEvent e) {
        fireTimer();
    }

    @Override
    public void changedUpdate(DocumentEvent e) {
        fireTimer();
    }
}

File: src/main/kotlin/mediathek/gui/dialogEinstellungen/allgemein/TimedDocumentListener.kt
package mediathek.gui.dialogEinstellungen.allgemein

import java.awt.event.ActionListener
import javax.swing.Timer
import javax.swing.event.DocumentEvent
import javax.swing.event.DocumentListener

/**
 * Listens to changes in a Document and finally fires the assigned Action.
 * All operations will be performed on Swing EDT.
 */
class TimedDocumentListener(taskPerformer: ActionListener) : DocumentListener {
    private val timer = Timer(1_000, taskPerformer).apply {
        isRepeats = false
    }

    private fun restartTimer() {
        timer.restart()
    }

    override fun insertUpdate(e: DocumentEvent) {
        restartTimer()
    }

    override fun removeUpdate(e: DocumentEvent) {
        restartTimer()
    }

    override fun changedUpdate(e: DocumentEvent) {
        restartTimer()
    }
}
